EIPS/eip-2200.md

```diff
@@ -0,0 +1,318 @@
+---
+eip: 2200
+title: Structured Definitions for Net Gas Metering
+author: Wei Tang (@sorpaas)
+discussions-to: https://github.com/sorpaas/EIPs/issues/1
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-07-18
+requires: 1283, 1706, 1884
+---
+
+## Simple Summary
+
+This is an EIP that implements net gas metering. It's a combined
+version of EIP-1283 and EIP-1706, with a structured definition so as
+to make it interoperable with other gas changes such as EIP-1884.
+
+## Abstract
+
+This EIP provides a structured definition of net gas metering changes
+for SSTORE opcode, enabling new usages for contract storage, and
+reducing excessive gas costs where it doesn’t match how most
+implementation works.
+
+This is a combination of EIP-1283 and EIP-1706.
+
+## Motivation
+
+This EIP proposes a way for gas metering on SSTORE, using information
+that is more universally available to most implementations, and
+require as little change in implementation structures as possible.
+
+* Storage slot’s original value.
+* Storage slot’s current value.
+* Refund counter.
+
+Usages that benefits from this EIP’s gas reduction scheme includes:
+
+* Subsequent storage write operations within the same call frame. This
+  includes reentry locks, same-contract multi-send, etc.
+* Exchange storage information between sub call frame and parent call
+  frame, where this information does not need to be persistent outside
+  of a transaction. This includes sub-frame error codes and message
+  passing, etc.
+
+The original definition of EIP-1283 created a danger of a new kind of
+reentrancy attacks on existing contracts as Solidity by default grants
```

Comment:
Nit: it's not Solidity's fault, it's part of the protocol, defined in YP

Reply:
Solidity sends the stipend if the call doesn't have value associated with it. The protocol only sets it if there's value sent.

Reply:
I'll mark this as resolve given this is just a nit and we have different opinions..